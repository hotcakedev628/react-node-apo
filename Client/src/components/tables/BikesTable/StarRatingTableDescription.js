import React from 'react'
import StarRatingComponent from 'react-star-rating-component';
import { connect } from 'react-redux';

const mapDispatchToProps = dispatch => ({ rateBike: (payload) => dispatch({ type: 'RATE_BIKE', payload }) })

export class Com extends React.Component {
    constructor(props) {
        super(props);
        this.state = { currentRate: props.rate };
    }

    render() {
        return (
            <td>
                <StarRatingComponent
                    onStarClick={(nextValue, prevValue, id) => this.props.rateBike({bikeId: id, rate:nextValue})} 
                    name={this.props.id}
                    value={this.state.currentRate || 0}
                    starCount={5}
                    onStarHover={(nextValue, prevValue, id) => this.setState({ currentRate: nextValue })}
                    onStarHoverOut={(nextValue, prevValue, id) => this.setState((oldState) => ({ currentRate: this.props.rate }))}
                />
                {
                    this.props.rate !== undefined ?
                        <p>
                            {this.props.rate} {this.props.rate >= 2 ? <span>  Stars  </span> : <span>  Star  </span>}
                        </p> :
                        <p> Not Rated yet </p>
                }
            </td>

        )
    }
}

export const StarRatingTableDescription = connect(null, mapDispatchToProps)(Com)
