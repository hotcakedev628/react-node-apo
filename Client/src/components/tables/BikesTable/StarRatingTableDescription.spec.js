import { StarRatingTableDescription } from './StarRatingTableDescription';
import renderer from 'react-test-renderer';
import React from 'react'



describe('StarRatingTableDescription', () => {
  fit('should match snapshot', () => {
    const tree = renderer.create(<StarRatingTableDescription 
        id="s"
        rate={4}
        onRateClick={()=>(<li>C</li>)}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
