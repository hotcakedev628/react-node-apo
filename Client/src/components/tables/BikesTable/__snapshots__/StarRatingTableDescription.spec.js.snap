// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRatingTableDescription should match snapshot 1`] = `
<td>
  <div
    className="dv-star-rating"
    style={
      Object {
        "display": "inline-block",
        "position": "relative",
      }
    }
  >
    <input
      checked={false}
      className="dv-star-rating-input"
      id="s_5"
      name="s"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
          "marginLeft": -9999,
          "position": "absolute",
        }
      }
      type="radio"
      value={5}
    />
    <label
      className="dv-star-rating-star dv-star-rating-empty-star"
      htmlFor="s_5"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#333",
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      <i
        style={
          Object {
            "fontStyle": "normal",
          }
        }
      >
        ★
      </i>
    </label>
    <input
      checked={true}
      className="dv-star-rating-input"
      id="s_4"
      name="s"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
          "marginLeft": -9999,
          "position": "absolute",
        }
      }
      type="radio"
      value={4}
    />
    <label
      className="dv-star-rating-star dv-star-rating-full-star"
      htmlFor="s_4"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffb400",
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      <i
        style={
          Object {
            "fontStyle": "normal",
          }
        }
      >
        ★
      </i>
    </label>
    <input
      checked={false}
      className="dv-star-rating-input"
      id="s_3"
      name="s"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
          "marginLeft": -9999,
          "position": "absolute",
        }
      }
      type="radio"
      value={3}
    />
    <label
      className="dv-star-rating-star dv-star-rating-full-star"
      htmlFor="s_3"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffb400",
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      <i
        style={
          Object {
            "fontStyle": "normal",
          }
        }
      >
        ★
      </i>
    </label>
    <input
      checked={false}
      className="dv-star-rating-input"
      id="s_2"
      name="s"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
          "marginLeft": -9999,
          "position": "absolute",
        }
      }
      type="radio"
      value={2}
    />
    <label
      className="dv-star-rating-star dv-star-rating-full-star"
      htmlFor="s_2"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffb400",
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      <i
        style={
          Object {
            "fontStyle": "normal",
          }
        }
      >
        ★
      </i>
    </label>
    <input
      checked={false}
      className="dv-star-rating-input"
      id="s_1"
      name="s"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
          "marginLeft": -9999,
          "position": "absolute",
        }
      }
      type="radio"
      value={1}
    />
    <label
      className="dv-star-rating-star dv-star-rating-full-star"
      htmlFor="s_1"
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "color": "#ffb400",
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      <i
        style={
          Object {
            "fontStyle": "normal",
          }
        }
      >
        ★
      </i>
    </label>
  </div>
  <p>
    4
     
    <span>
        Stars  
    </span>
  </p>
</td>
`;
